crates/stages/src/stages/merkle.rs

```diff
@@ -5,28 +5,44 @@ use reth_db::database::Database;
 use std::fmt::Debug;
 use tracing::*;
 
-const MERKLE_EXECUTION: StageId = StageId("MerkleExecuteStage");
-const MERKLE_UNWIND: StageId = StageId("MerkleUnwindStage");
+/// The [`StageId`] of the merkle hashing execution stage.
+pub const MERKLE_EXECUTION: StageId = StageId("MerkleExecuteStage");
 
-/// Merkle stage uses input from [AccountHashingStage] and [StorageHashingStage] stages
-/// and calculated intermediate hashed and state root.
-/// This stage depends on the Account and Storage stages. It will be executed after them during
-/// execution, and before them during unwinding.
+/// The [`StageId`] of the merkle hashing unwind stage.
+pub const MERKLE_UNWIND: StageId = StageId("MerkleUnwindStage");
+
+/// The merkle hashing stage uses input from
+/// [`AccountHashingStage`][crate::stages::AccountHashingStage] and
+/// [`StorageHashingStage`][crate::stages::AccountHashingStage] to calculate intermediate hashes
+/// and state roots.
+///
+/// This stage should be run with the above two stages, otherwise it is a no-op.
+///
+/// This stage is split in two: one for calculating hashes and one for unwinding. TODO: Why?
```

Comment:
(cc @rakita?) I am not sure why this is split into an execution and an unwind part, but it should be in the docs

Reply:
Where is best place to put the docs? merkle stage depends on Hashing stage, it needs to come before Hashing stages for both execution and unwind paths (unwind goes in backwards), this solves unwind ordering that is present in erigon.

Reply:
It should be in this rustdoc, my question is more precisely: Why does it need to unwind before the hashing stages?